force-app/main/default/lwc/accountMap/__tests__/accountMap.test.js

```diff
@@ -0,0 +1,80 @@
+
+// Register as a LDS wire adapter. Some tests verify the provisioned values trigger desired behavior.
+const getRecordAdapter = registerLdsTestWireAdapter(getRecord);
+describe('c-account-map', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+    });
+
+    it('displays a lightning-map when the wire adaptor returns records', () => {
+        // Create element
+        const element = createElement('c-account-map', {
+            is: AccountMap
+        });
+        document.body.appendChild(element);
+
+        // Emit data from the getRecord adapter.
+        getRecordAdapter.emit(mockGetRecord);
+
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise rejects.
+        return Promise.resolve().then(() => {
+            // Select elements for validation
+            let mapEl = element.shadowRoot.querySelector('lightning-map');
+            expect(mapEl.length).toBe(mockGetRecord.length);
+        });
+    });
+
+    it('displays an error panel when the wire adaptor returns an empty array', () => {
+        // Create element
+        const element = createElement('c-account-map', {
+            is: AccountMap
+        });
+        document.body.appendChild(element);
+
+        // Emit an empty array from the getRecord adapter.
+        getRecordAdapter.emit([]);
+
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise rejects.
+        return Promise.resolve().then(() => {
+            // Select elements for validation
+            let mapEl = element.shadowRoot.querySelector('lightning-map');
+            expect(mapEl.length).toBe(mockGetRecord.length);
+        });
+    });
+
+    it('displays an error panel when the Apex wire adaptor returns an error', () => {
+        // Create element
+        const element = createElement('c-account-map', {
+            is: AccountMap
+        });
+        document.body.appendChild(element);
+
+        // Emit an error from the getRecord adapter.
+        getRecordAdapter.error();
+
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise rejects.
+        return Promise.resolve().then(() => {
+            // Select elements for validation
+            const errorPanelEl = element.shadowRoot.querySelector(
+                'c-error-panel'
+            );
+            expect(errorPanelEl).not.toBeNull();
```

Comment:
A nice addition would be to also check if `errroPanelEl.errors` is set, to make sure that `errors` is passed correctly.

Reply:
Hey @muenzpraeger @pozil - Is this really something that needs to be tested here? Shouldn't functionality of errorPanel be tested in the definition of the errorPanel component?

Reply:
The assert would check if your component actually passes its errors to the error panel, so that it can render the errors.

Reply:
Ah, yes, thanks for the clarification!